index.js

```diff
@@ -77,10 +79,23 @@ export default class Purchases {
   */
 
   /** Sets a function to be called on purchase complete or fail
-      @param {PurchaseListener} restoreTransactionsListener_ Purchase listener 
+      @param {PurchaseListener} purchaseListener_ Purchase listener 
+      @returns {String} id Purchase listener id for future removal
   */
   static addPurchaseListener(purchaseListener_) {
-    purchaseListener.push(purchaseListener_);
+    if (typeof purchaseListener_ !== "function") {
+      throw new Error("addPurchaseListener needs a function")
+    }
+    const id = ID();
+    purchaseListeners[id] = purchaseListener_;
+    return id;
+  }
+
+  /** Removes a given Purchase listener
+      @param {String} id Purchase listener id
```

Comment:
we could add here `@returns true` to avoid confusion, like this will always return true whether or not the id exists. 

Alternatively we could check the id and return false in that case.